source/common/http/conn_manager_impl.cc

```diff
@@ -278,7 +278,20 @@ void ConnectionManagerImpl::doDeferredStreamDestroy(ActiveStream& stream) {
 
   stream.completeRequest();
   stream.filter_manager_.onStreamComplete();
-  stream.filter_manager_.log();
+
+  // For HTTP/3, skip access logging here and add deferred logging info
+  // to stream info for QuicStatsGatherer to use later. If there was a
+  // downstream reset, log here as usual.
+  if (codec_ && codec_->protocol() == Protocol::Http3 &&
+      !stream.filter_manager_.sawDownstreamReset() &&
+      Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.quic_defer_logging_to_ack_listener")) {
+    stream.filter_manager_.streamInfo().setDeferredLoggingHeadersAndTrailers(
+        stream.DeferredLoggingHeadersAndTrailers());
```

Comment:
Can you wrap this logic into into ActiveStream::deferLoggingHeadersAndTrailers()?

Reply:
Done.